tiledb/ml/readers/pytorch.py

```diff
@@ -98,10 +97,11 @@ def __iter__(self) -> Iterator[DataType]:
 
         # Shuffle offsets in case we need batch shuffling
         if self.batch_shuffle:
-            random.seed()
-            offsets = random.sample(  # type: ignore
-                range(iter_start, iter_end, self.batch_size),
+            gen = np.random.default_rng()
```

Comment:
The previous comment about `seed()` applies here too; the point is to _not_ create a new random generator every time this is called. If there is a need for explicit generator it should be created once in `__init__`, otherwise `np.random.choice` is good enough.